components/infobox/wikis/brawlhalla/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,144 @@
+	['commentator'] = 'Commentator', ['caster'] = 'Commentator',
+	['expert'] = 'Analyst', ['host'] = 'Host', ['streamer'] = 'Streamer',
+	['interviewer'] = 'Interviewer', ['journalist'] = 'Journalist',
+	['manager'] = 'Manager', ['map maker'] = 'Map maker',
+	['observer'] = 'Observer', ['photographer'] = 'Photographer',
+	['tournament organizer'] = 'Organizer', ['organizer'] = 'Organizer',
+}
+local _CLEAN_OTHER_ROLES = {
+	['coach'] = 'Coach', ['staff'] = 'false',
+	['content producer'] = 'Content producer', ['streamer'] = 'false',
+}
+
+local _CURRENT_YEAR = tonumber(os.date('%Y'))
+local _statusStore
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _earningsPerYear = {}
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.getStatusToStore = CustomPlayer.getStatusToStore
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.getPersonType = CustomPlayer.getPersonType
+
+	player.calculateEarnings = CustomPlayer.calculateEarnings
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'role' then return {}
+	elseif
+		id == 'history' and
+		string.match(_args.retired or '', '%d%d%d%d')
+	then
+		table.insert(widgets, Cell{
+				name = 'Retired',
+				content = {_args.retired}
+			})
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local yearsActive = ActiveYears.display{
+		player = _PAGENAME,
+	}
+
+	local currentYearEarnings = _earningsPerYear[tostring(_CURRENT_YEAR)]
+	if currentYearEarnings then
+		currentYearEarnings = Math.round{currentYearEarnings}
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
+
+	return {
+		Cell{
+			name = 'Approx. Winnings ' .. _CURRENT_YEAR,
+			content = {currentYearEarnings}
+		},
+		Cell{name = 'Years active', content = {yearsActive}},
+		Cell{name = 'Main Legends', content = {PlayersSignatureLegends.get{
+			player = _PAGENAME,
+		}}},
+	}
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:adjustLPDB(lpdbData, _, personType)
+	local extradata = lpdbData.extradata
+	extradata.lc_id = string.lower(self.pagename)
+	extradata.teamname = _args.team
+	extradata.role = _args.role
+	extradata.role2 = _args.role2
+	extradata.activeplayer = (not _statusStore) and Variables.varDefault('isActive', '') or ''
+
+	lpdbData.extradata = extradata
+
+	return lpdbData
+end
+
+function CustomPlayer:calculateEarnings(args)
+	local totalEarnings
+	totalEarnings, _earningsPerYear = Earnings.calculateForPlayer{
+		player = args.earnings or self.pagename,
+		perYear = true
+	}
+	return totalEarnings, _earningsPerYear
+end
```

Comment:
Isn't this what commons already does?

Reply:
getting `attempt to index global '_earningsPerYear' (a nil value)` error if i remove it